manifests/base/webhook/manifests.yaml

```diff
@@ -0,0 +1,26 @@
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: validating-webhook-configuration
```

Comment:
We can not modify this name since this manifests is generated by the controller-gen, and the controller-gen doesn't have any knob to modify this name: https://book.kubebuilder.io/reference/markers/webhook

Reply:
@tenzen-y Can we use this kubebuilder param to the the name for VWC ?
 
<img width="783" alt="Screenshot 2024-04-08 at 12 51 38" src="https://github.com/kubeflow/training-operator/assets/31112157/0741a382-0009-4f6a-baa3-380c0a337e8e">

Reply:
We already add a prefix to VWN name here: https://github.com/kubeflow/training-operator/blob/7cc1365c5aac01beef660709a5139321a5e62675/manifests/base/webhook/kustomization.yaml#L6

Reply:
@tenzen-y That's right, what do you think about adding additional [patch](https://github.com/kubeflow/training-operator/blob/7cc1365c5aac01beef660709a5139321a5e62675/manifests/base/webhook/patch.yaml) to name it as `training-operator.kubeflow.org` to make it consistent with Katib and also how it is suggested in K8s doc: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#configure-admission-webhooks-on-the-fly?

Reply:
I confirmed that we can apply patch `validator.training-operator.kubeflow.org` to VWC locally.

I will put the commit.

Reply:
Great, thank you!